src/c-opentimelineio/opentimelineio-bindings/include/copentimelineio/typeRegistry.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "errorStatus.h"
+#include "typeInfo.h"
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+    struct TypeRegistry;
+    typedef struct TypeRegistry TypeRegistry;
+    TypeRegistry*               TypeRegistry_instance();
+    _Bool                       TypeRegistry_register_type(
+                              TypeRegistry* self,
+                              const char*   schema_name,
+                              int           schema_version,
+                              TypeInfo*     type,
+                              function*     create,/*how to pass such functions*/
```

Comment:
Even if we wrap std::function like we've done with the other classes, is there a way we can pass it from C to C++? I don't think we can create such function pointers in C.

Reply:
Also without being able to use classes and therefore not being able to extend to them, would it be possible to define schemas in C?

Reply:
In C, the idiomatic way to deal with function pointers is to typedef them.

e.g. ```typedef int (*FuncPtr)(int, int); FuncPtr ptr = &ActualFunction;```

There are common approaches to schemas and classes in C, c.f. COM. The larger question is how they would be used in the C OTIO API. Perhaps a good next exercise would be to write a little snip of code that exercises the registry in a hypothetical way, rather than strictly wrapping the C++ interfaces. That would give us a discussion point.

Reply:
> schemas and classes in C, c.f. COM

What is COM? :sweat_smile: 

I still don't have a proper idea of how the type registry works and how exactly the schemas are defined. Are there tests/examples of using the registry with the C++ API. The docs only focuses on python.